packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -77,7 +79,11 @@ export interface SdkOperationGroup {
   service: Namespace;
 }
 
-interface SdkTypeBase {
+interface DecoratedType {
```

Comment:
is type a valid use case?

Reply:
we can always add things, should start with smaller scope we are sure of

Reply:
this is somehow same pattern with typespec core. since lots of type will have decorators, i added it as a separated interface.